v3/src/components/graph/models/graph-model.ts

```diff
@@ -117,7 +113,13 @@
     removeAxis(place: AxisPlace) {
       self.axes.delete(place)
     },
-    setAttributeID(role: GraphAttrRole, id: string) {
+    setAttributeID(role: GraphAttrRole, dataSetID: string, id: string) {
+      const dataSet = getDataSetFromId(self, dataSetID)
+      if (dataSet && isTileLinkedToOtherDataSet(self, dataSet)) {
```

Comment:
Yes, this is the right thing to do. But I don't know how it will interact with the GraphController's handleAttributeAssignment. Maybe OK?

Reply:
See comment in `GraphController` for response.